runelite-client/src/main/java/net/runelite/client/ui/components/DimmableJPanel.java

```diff
@@ -0,0 +1,82 @@
+package net.runelite.client.ui.components;
+
+import java.awt.Color;
+import javax.swing.JPanel;
+
+public class DimmableJPanel extends JPanel
+{
+	// Dimming state, allows for restoring original colors before dimming
+	private boolean dimmed = false;
+	private Color dimmedForeground = null;
+	private Color dimmedBackground = null;
+	private Color undimmedForeground = null;
+	private Color undimmedBackground = null;
+
+	@Override
+	public void setForeground(Color color)
+	{
+		undimmedForeground = color;
+		dimmedForeground = color.darker();
+		super.setForeground(color);
+	}
+
+	@Override
+	public void setBackground(Color color)
+	{
+		undimmedBackground = color;
+		dimmedBackground = color.darker();
+		super.setBackground(color);
+	}
+
+	@Override
+	public Color getForeground()
+	{
+		return dimmed ? dimmedForeground : undimmedForeground;
+	}
+
+	@Override
+	public Color getBackground()
+	{
+		return dimmed ? dimmedBackground : undimmedBackground;
+	}
+
+	/**
+	 * Dimming sets all parts of this component with darker colors except for the central label
+	 * This is useful for showing that progress is paused
+	 * Setting dim to false will restore the original colors from before the component was dimmed.
+	 * @param dimmed
+	 */
+	public final void setDimmed(boolean dimmed)
+	{
+		if (this.dimmed == dimmed)
+		{
+			// No change to be made
+			return;
+		}
+
+		this.dimmed = dimmed;
+
+		if (dimmed)
+		{
+			super.setBackground(dimmedBackground);
+			super.setForeground(dimmedForeground);
+		}
+		else
+		{
+			super.setBackground(undimmedBackground);
+			super.setForeground(undimmedForeground);
+		}
+		// Allow user behavior here
+		onDimmedChanged(dimmed);
```

Comment:
I dont think this is really necessary, as user using this class can just override this method and call super.

Reply:
In this case, I am hiding that I deduplicate the dim state changes. I don't want extenders to keep redundant state.

Reply:
I see, I dont think that should really be concern here or anywhere else as the setBack/Foreground is not expensive at all so even you dont need to handle this state